File: src/net/merayen/elastic/ui/objects/top/views/arrangementview/TrackList.java
package net.merayen.elastic.ui.objects.top.views.arrangementview;

import net.merayen.elastic.ui.UIObject;
import net.merayen.elastic.ui.objects.components.autolayout.AutoLayout;
import net.merayen.elastic.ui.objects.components.autolayout.LayoutMethods;

class TrackList extends AutoLayout {
	public float width;

	public TrackList() {
		super(new LayoutMethods.HorizontalBox());
	}

	@Override
	protected void onUpdate() {
		super.onUpdate();
		for(UIObject obj : search.getChildren()) {
			Track track = ((Track)obj);
			track.width = width - 10;
		}
	}
}

File: src/main/kotlin/net/merayen/elastic/ui/objects/top/views/arrangementview/TrackList.kt
package net.merayen.elastic.ui.objects.top.views.arrangementview

import net.merayen.elastic.ui.FlexibleDimension
import net.merayen.elastic.ui.objects.components.autolayout.AutoLayout
import net.merayen.elastic.ui.objects.components.autolayout.LayoutMethods

internal class TrackList : AutoLayout<LayoutMethods.HorizontalBox>(LayoutMethods.HorizontalBox()), FlexibleDimension {
    override var layoutWidth = 100f
    override var layoutHeight = 50f

    override fun onUpdate() {
        super.onUpdate()
        for (obj in search.children)
            (obj as TrackPane).layoutWidth = layoutWidth

        placement.maxWidth = layoutWidth
    }
}
